test/ReaderTest.cpp

```diff
@@ -326,4 +326,16 @@ BOOST_AUTO_TEST_CASE(TestReadNonExistingFile)
                           check_msg);
 }
 
+BOOST_AUTO_TEST_CASE(TestReaderEof)
+{
+    Reader r(uncompressed_sample);
+
+    size_t count = 0;
+    while (!r.eof()) {
```

Comment:
Why is this preferred to the syntax in `BOOST_AUTO_TEST_CASE(TextFileReader)`?

How is it better than the syntaxes already available?

    // current demos (e.g. in BOOST_AUTO_TEST_CASE(TextFileReader))
    std::shared_ptr<Block> b;
    while ((b = r.next(CT_BLOCK)) != nullptr) {
        // do stuff with block
    }

    // proposed syntax:
    while (!r.eof()) {
        auto b = r.next(CT_BLOCK);
    }

    // alternate syntax: 
    for (auto b = r.next(CT_BLOCK); b != nullptr; b = r.next(CT_BLOCK)) {

    }

If this is intended to avert problems when reading broken files, maybe the test should use a broken/unreadable file, and all the examples should be updated to use the recommended syntax.

Reply:
The test does not intend to present a preferred syntax, but just to exercise the new `eof()` method, checking that, as long as EOF is not detected, the next CT block can be retrieved, and that EOF will be reached after visitation of the known number of structures in the file (3).

The intention of the new method is providing information about the availability of data without actually trying to parse that data in advance.